internal/http/services/owncloud/ocgraph/shares.go

```diff
@@ -0,0 +1,415 @@
+// Copyright 2018-2024 CERN
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// In applying this license, CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+// This package implements the APIs defined in https://owncloud.dev/apis/http/graph/spaces/
+
+package ocgraph
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"path"
+	"strings"
+
+	gateway "github.com/cs3org/go-cs3apis/cs3/gateway/v1beta1"
+	groupv1beta1 "github.com/cs3org/go-cs3apis/cs3/identity/group/v1beta1"
+	userv1beta1 "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+
+	collaborationv1beta1 "github.com/cs3org/go-cs3apis/cs3/sharing/collaboration/v1beta1"
+	linkv1beta1 "github.com/cs3org/go-cs3apis/cs3/sharing/link/v1beta1"
+	provider "github.com/cs3org/go-cs3apis/cs3/storage/provider/v1beta1"
+	"github.com/cs3org/reva/pkg/appctx"
+	"github.com/cs3org/reva/pkg/spaces"
+	"github.com/cs3org/reva/pkg/utils"
+	libregraph "github.com/owncloud/libre-graph-api-go"
+)
+
+func (s *svc) getSharedWithMe(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := appctx.GetLogger(ctx)
+
+	gw, err := s.getClient()
+	if err != nil {
+		log.Error().Err(err).Msg("error getting grpc client")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	resShares, err := gw.ListExistingReceivedShares(ctx, &collaborationv1beta1.ListReceivedSharesRequest{})
+	if err != nil {
+		log.Error().Err(err).Msg("error getting received shares")
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	shares := make([]*libregraph.DriveItem, 0, len(resShares.ShareInfos))
+	for _, share := range resShares.ShareInfos {
+		drive, err := s.cs3ReceivedShareToDriveItem(ctx, share)
+		if err != nil {
+			log.Error().Err(err).Msg("error getting received shares")
+			w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Let's not accumulate further technical debt with those HTTP 500:
* On a loop, we should skip and return the rest
* In the other cases, we should return something relevant for the frontend to show (and help us in debugging afterwards)